### PR TITLE
Convert build process from Travis-CI to GitHub Actions

### PR DESCRIPTION
**Pull request by dcasavant**
The purpose of this change is to switch the build process from Travis-CI to GitHub Actions. Actions offer several advantages over the current Travis-CI capabilities:
* They are free.
* They are easier to access because their security is repo-integrated rather than via separate website.
* They are better integrated with GitHub.

As this is a change to the build process, additional review and refinement is required before merge.

### CODE REVIEW

File: .github/workflows/docker-build.yml [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+        path: ${{ steps.scan.outputs.sarif }}
+        
+    - name: Upload Anchore Scan SARIF Report
+      uses: github/codeql-action/upload-sarif@v1
+      with:
+        sarif_file: ${{ steps.scan.outputs.sarif }}
+
+  push:
+    name: Push to DockerHub
+    if: ${{ github.ref == 'refs/heads/main' && github.repository == 'pegasystems/docker-pega-web-ready' }}
```
Comment (kishorv10) [inline .github/workflows/docker-build.yml:100]: Current repository has master branch. Its ref might be `refs/heads/master`?

File: .github/workflows/docker-build.yml [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+    runs-on: ubuntu-latest
+    needs: [scan, test]
+    steps:
+      - run: make push
```
Comment (kishorv10) [inline .github/workflows/docker-build.yml:104]: I tried Github actions workflow on my forked repository [here](https://github.com/kishorv10/docker-pega-web-ready/blob/master/.github/workflows/docker-build.yml). This stage needs more steps, code checkout and docker login. Below sets of actions worked for me.
```
steps:
      - name: Login to Docker Hub
        uses: docker/login-action@v1
        with:
          username: ${{ secrets.DOCKERHUB_USERNAME }}
          password: ${{ secrets.DOCKERHUB_TOKEN }}
      - name: Check out working repository
        uses: actions/checkout@v2
      - name: Push image to dockerhub 
        run: make push
 ```

**Review by kishorv10 — CHANGES_REQUESTED** [review]
@dcasavant docker scan is failing with a higher CVE reported by anchore (cve-2015-8751). We might need to fix it before we make this shift. I will create an internal bug item for it.
In general changes look good, more steps needed to be added to docker push stage.

File: .github/workflows/docker-build.yml (hunk 1 above)
Comment (dcasavant) [reply .github/workflows/docker-build.yml]: updated

File: .github/workflows/docker-build.yml (hunk 2 above)
Comment (dcasavant) [reply .github/workflows/docker-build.yml]: updated

**Comment by dcasavant** [comment]
Closing this PR and reopening post-conflict resolution

**Pull request closed without merging**